packages/storefront-api-client/src/tests/storefront-api-client.test.ts

```diff
@@ -0,0 +1,471 @@
+        });
+
+        it("throws an error when the api version is not provided", () => {
+          expect(() =>
+            createStorefrontAPIClient({
+              ...config,
+              // @ts-ignore
+              apiVersion: undefined,
+            })
+          ).toThrow(
+            new Error(
+              `Storefront API Client: the provided \`apiVersion\` is invalid. Current supported API versions: ${mockApiVersions.join(
+                ", "
+              )}`
+            )
+          );
+        });
+
+        it("throws an error when the api version is not a string", () => {
+          expect(() =>
+            createStorefrontAPIClient({
+              ...config,
+              // @ts-ignore
+              apiVersion: { year: 2022, month: 1 },
+            })
+          ).toThrow(
+            new Error(
+              `Storefront API Client: the provided \`apiVersion\` is invalid. Current supported API versions: ${mockApiVersions.join(
+                ", "
+              )}`
+            )
+          );
+        });
+
+        it("console warns when a unsupported api version is provided", () => {
+          const consoleWarnSpy = jest
+            .spyOn(window.console, "warn")
+            .mockImplementation(jest.fn());
+
+          createStorefrontAPIClient({
+            ...config,
+            apiVersion: "2022-07",
+          });
+
+          expect(consoleWarnSpy).toHaveBeenCalledWith(
+            `Storefront API Client: the provided \`apiVersion\` (\`2022-07\`) is deprecated or not supported. Current supported API versions: ${mockApiVersions.join(
+              ", "
+            )}`
+          );
+        });
+
+        it("throws an error when neither public and private access tokens are provided", () => {
+          expect(() =>
+            // @ts-ignore
+            createStorefrontAPIClient({
+              ...config,
+              publicAccessToken: undefined,
+            })
+          ).toThrow(
+            new Error(
+              `Storefront API Client: a public or private access token must be provided`
+            )
+          );
+        });
+
+        it("throws an error when a private access token is provided in a browser environment (window is defined)", () => {
+          expect(() =>
+            // @ts-ignore
+            createStorefrontAPIClient({
+              ...config,
+              publicAccessToken: undefined,
+              privateAccessToken: "private-access-token",
+            })
+          ).toThrow(
+            new Error(
+              "Storefront API Client: private access tokens and headers should only be used in a server-to-server implementation. Use the API public access token in nonserver environments."
+            )
+          );
+        });
+
+        it("throws an error when both public and private access tokens are provided in a server environment (window is undefined)", () => {
+          const windowSpy = jest
+            .spyOn(window, "window", "get")
```

Comment:
Why do you spy here?

Reply:
This is to simulate a server environment - the condition of being able to provide both a public and private token can only happen in a server environment and if we didn't spy on this, the earlier check for the private token only allowed in non browser scenarios will fire and we can't never reach this error message.